src/modules/exposure-period/exposure-period.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable, InternalServerErrorException, Logger } from '@nestjs/common';
+import { InjectDataSource } from '@nestjs/typeorm';
+import { DataSource } from 'typeorm';
+
+import { ExposurePeriodDto } from './dto/exposure-period.dto';
+
+@Injectable()
+export class ExposurePeriodService {
+  private readonly logger = new Logger();
+
+  constructor(
+    @InjectDataSource('mssql-mdm')
+    private readonly mdmDataSource: DataSource,
+  ) {}
+
+  async calculate(startDate: Date, endDate: Date, productGroup: string): Promise<ExposurePeriodDto> {
+    try {
+      // TODO: SP USP_MDM_READ_EXPOSURE_PERIOD is not using data/tables from DB. Calculation could be moved to Javascript.
+      const spResults = await this.mdmDataSource.query('USP_MDM_READ_EXPOSURE_PERIOD @0, @1, @2', [startDate, endDate, productGroup]);
+
+      if (!spResults || !spResults[0] || typeof spResults[0].EXPOSURE_PERIOD === 'undefined') {
+        throw new InternalServerErrorException('No exposure period result from USP_MDM_READ_EXPOSURE_PERIOD');
```

Comment:
should this be a 404 instead? rather than 500 type of error?

Reply:
If input is not correct it is 400.
If input is correct it should always return calculation. Simplified calculation is endDate - startDate in months (rounded up to full month). 
I think 500 is better code in case calculation doesn't have result and something is wrong on server side.

Reply:
makes sense